lib/datasource/npm/get.ts

```diff
@@ -78,18 +77,8 @@ export async function getDependency(
 
   const uri = url.parse(packageUrl);
 
-  if (uri.host === 'registry.npmjs.org' && !uri.pathname.startsWith('/@')) {
-    // Delete the authorization header for non-scoped public packages to improve http caching
-    // Otherwise, authenticated requests are not cacheable until the registry adds "public" to Cache-Control
-    // Ref: https://greenbytes.de/tech/webdav/rfc7234.html#caching.authenticated.responses
-    delete headers.authorization;
-  }
```

Comment:
We now do datasource-level caching anyway, so I don't think this benefits us at all

Reply:
The refactoring of this logic is why we have some extra `authorization` headers in snapshots of this PR